File: src/api/MaterialAdvisor/MaterialAdvisor.Application/Models/KnowledgeChecks/UnverifiedAnswer.cs
using MaterialAdvisor.Application.Models.Topics;
using MaterialAdvisor.Application.Models.Users;

namespace MaterialAdvisor.Application.Models.KnowledgeChecks;

public class UnverifiedAnswer
{
    public SubmittedAnswer SubmittedAnswer { get; set; } = null!;

    public User User { get; set; } = null!;

    public KnowledgeCheckTopicListItem KnowledgeCheck { get; set; } = null!;

    public Topic Topic { get; set; } = null!;

    public ICollection<VerifiedAnswer> VerifiedAnswers { get; set; } = [];
}

File: src/app/material-advisor/src/app/models/knowledge-check/UnverifiedAnswer.ts
import { SubmittedAnswer } from "./SubmittedAnswer";
import { Topic } from "@models/topic/Topic";
import { KnowledgeCheckTopicListItem } from "./KnowledgeCheckTopicListItem";
import { User } from "@shared/models/User";
import { VerifiedAnswer } from "./VerifiedAnswer";

export interface UnverifiedAnswer {
  verifiedAnswers: VerifiedAnswer[];
  submittedAnswer: SubmittedAnswer;
  topic: Topic;
  knowledgeCheck: KnowledgeCheckTopicListItem;
  user: User;
}